src/app/components/shared/datetime-formats/datetime/abstract-datetime.component.ts

```diff
@@ -65,11 +71,11 @@ export abstract class AbstractDatetimeComponent extends AbstractTemplateComponen
   }
 
   private dateTimeFormat(): string {
-    if (this.date && this.time) {
+    if (this.date() && this.time()) {
       return AbstractDatetimeComponent.FULL_DATETIME;
-    } else if (this.date) {
+    } else if (this.date()) {
       return AbstractDatetimeComponent.DATE_ONLY;
-    } else if (this.time) {
+    } else if (this.time()) {
```

Comment:
ditto, don't execute the methods more than once

Reply:
Chosen variable names: `showDate` and `showTime`

I chose this as to reduce the impact of typos eg. if we used `const date`, if we forgot to type `this` we would be using the wrong variable.

Additionaly, in the previous PR, we agreed that `showDate` and `showTime` were semantically correct, however, to improve dx, we shortened the attribute names to `date` and `time`